File: ui/src/models/Process.ts
export interface ProcessIdentity {
  path: string
  signer: string
  company: string
  category: string
  startTime?: number
  riskScore: number
}

export interface Process {
  pid: number
  name: string
  parentPid: number
  cpuPercent: number
  memoryMb: number
  threads: number
  identity?: ProcessIdentity
  children: number[]
  startTime: number
  riskScore: number
}

File: src/core/process.rs
#[derive(Debug, Clone)]
pub struct Process {
    pub pid: u32,
    pub name: String,
    pub cpu_percent: f32,
    pub memory_mb: f32,
    pub threads: u32,
    pub parent_pid: u32,
    pub start_time: Option<u64>,
}
